recommend/runtimePolicy.go

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package recommend
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	opb "github.com/accuknox/auto-policy-discovery/src/protobuf/v1/observability"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const serviceAccData = "/run/secrets/kubernetes.io/serviceaccount"
+
+// CreateRuntimePolicy function generates runtime policy for service account
+func CreateRuntimePolicy(o *Options, img *ImageInfo) error {
+	var label string
+
+	for _, labels := range o.Labels {
+		kvPair := strings.FieldsFunc(labels, labelSplitter)
+		label = strings.TrimPrefix(fmt.Sprintf("%s,%s=%s", label, kvPair[0], kvPair[1]), ",")
+
+	}
+
+	err := summarySearch(label, img)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+func summarySearch(labels string, img *ImageInfo) error {
+	ms := MatchSpec{}
+
+	gRPC := ""
+
+	if val, ok := os.LookupEnv("DISCOVERY_SERVICE"); ok {
+		gRPC = val
+	} else {
+		gRPC = "localhost:9089"
+	}
+
+	// create a client
+	conn, err := grpc.Dial(gRPC, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return errors.New("could not connect to the server. Possible troubleshooting:\n- Check if discovery engine is running\n- Create a portforward to discovery engine service using\n\t\033[1mkubectl port-forward -n explorer service/knoxautopolicy --address 0.0.0.0 --address :: 9089:9089\033[0m\n[0m")
+	}
+	defer conn.Close()
+
+	client := opb.NewObservabilityClient(conn)
+
+	podResp, err := client.GetPodNames(context.Background(), &opb.Request{
+		Label:     labels,
+		NameSpace: img.Namespace,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	for _, podName := range podResp.PodName {
+		sumResp, err := client.Summary(context.Background(), &opb.Request{
+			PodName: podName,
+			Type:    "file",
+		})
+		if err != nil {
+			return err
+		}
+		ms, err = checkProcessFileData(sumResp, ms)
+		if err != nil {
+			return err
+		}
+		img.writePolicyFile(ms)
+
+	}
+
+	return nil
+
+}
+
+func createRuntimePolicy(filePaths SysRule, ms MatchSpec) MatchSpec {
+	ms.Rules = Rules{
+		FileRule: &filePaths,
+	}
+
+	ms.OnEvent.Severity = 1
+	ms.Description.Tldr = "Kubernetes serviceaccount folder access should be limited "
+	ms.Precondition = serviceAccData
+	ms.Name = "audit-serviceaccount-runtime"
+	ms.OnEvent.Tags = []string{"KUBERNETES", "SERVICE ACCOUNT", "RUNTIME POLICY"}
+	ms.OnEvent.Message = "serviceaccount access detected"
+	ms.OnEvent.Action = "Audit"
+
+	return ms
+
+}
+
+func checkProcessFileData(sumResp *opb.Response, ms MatchSpec) (MatchSpec, error) {
```

Comment:
Why this function receives an empty `ms` just to carry it forward to another function?
1. Let's remove `ms` argument from both `createProcessFIleData()` and `createRuntimePolicy()`. 
2. Please make `createRuntimePolicy()` create new `ms` locally with all filled value and return it.